src/backend/partaj/core/api/referral_attachment.py

```diff
@@ -0,0 +1,114 @@
+    be implemented by ViewSets using this permission for objects with a relation to a referral.
+    """
+
+    def has_permission(self, request, view):
+
+        referral = view.get_referral(request)
+        return (
+            request.user.is_authenticated
+            and referral.users.filter(users__id=request.user.id).exists()
+        )
+
+
+class ReferralAttachmentViewSet(viewsets.ModelViewSet):
+    """
+    API endpoints for referral answer attachments.
+    """
+
+    permission_classes = [NotAllowed]
+    queryset = models.ReferralAnswerAttachment.objects.all()
+    serializer_class = ReferralAttachmentSerializer
+
+    def get_permissions(self):
+        """
+        Manage permissions for default methods separately, delegating to @action defined
+        permissions for other actions.
+        """
+        if self.action in ["create", "retrieve"]:
+            permission_classes = [UserIsReferralRequester]
+        else:
+            try:
+                permission_classes = getattr(self, self.action).kwargs.get(
+                    "permission_classes"
+                )
+            except AttributeError:
+                permission_classes = self.permission_classes
+
+        return [permission() for permission in permission_classes]
+
+    def get_referral(self, request):
+        """
+        Helper: get the related referral, return an error if it does not exist.
+        """
+        referral_id = request.data.get("referral") or request.query_params.get(
+            "referral"
+        )
+        try:
+            referral = models.Referral.objects.get(id=referral_id)
+        except models.Referral.DoesNotExist as error:
+            raise Http404(
+                f"Referral {request.data.get('referral')} not found"
+            ) from error
+
+        return referral
+
+    def create(self, request, *args, **kwargs):
+        """
+        Let users create new referral  attachment, processing the file itself along with
+        its metadata to create a ReferralAttachment instance.
+        """
+        # Make sure the referral  exists and return an error otherwise.
+        referral = self.get_referral(request)
+
+        if len(request.FILES.getlist("files")) > 1:
+            return Response(
+                status=400,
+                data={
+                    "errors": [
+                        "Referral  attachments cannot be created with more than one file."
+                    ]
+                },
+            )
+
+        try:
+            file = request.FILES.getlist("files")[0]
+        except IndexError:
+            return Response(
+                status=400,
+                data={
+                    "errors": [
+                        "Referral  attachments cannot be created without a file."
+                    ]
+                },
+            )
+
+        attachment = models.ReferralAttachment.objects.create(
```

Comment:
And you then have a problem here: the referral needs to be set at the time when you create the `ReferralAttachment` as this is a non-nullable `ForeignKey`. On referral answers, the similar field is a `ManyToMany`, which is by definition nullable.